File: trunk/Client/Lua/PanelLogon.lua
--PanelLogon
print("PanelLogon has loaded.")
local LuaPanel = Assets.Script.Frame.LuaPanel
PanelLogon = {}
setmetatable(PanelLogon, LuaPanel)
PanelLogon.__index = PanelLogon

function PanelLogon.New(base)
	print("PanelLogon.New")
	o = {}
	o.base = base;
	print(o.base.name)
	return o
end

function PanelLogon:Init()
	Debugger.Log("LuaPanelLogon.Init")
	Debugger.Log(self.base.name)
	Debugger.Log(self.base.controls)

	self.username = self.base.controls:GetInputField("InputFieldUsername")
	self.password = self.base.controls:GetInputField("InputFieldPassword")
	Debugger.Log(self.username);
end

function PanelLogon:OnButtonLogonClick()
	Debugger.Log("Lua_PanelLogon.OnButtonLogonClick")
	Debugger.Log(self.username);
	Debugger.Log("username:"..self.username.text);
	Debugger.Log("password:"..self.password.text);

	reqLogon = ReqLogon.New();
    reqLogon.Username = self.username.text;
    reqLogon.Password = self.password.text;
    Net.me:Send2(reqLogon);
end

function PanelLogon:OnButtonRegistClick()
	Debugger.Log("Lua_PanelLogon.OnButtonLogonClick")
	Debugger.Log("username:"..self.username.text);
	Debugger.Log("password:"..self.password.text);

	reqRegist = ReqRegist.New();
	reqRegist.Username = self.username.text;
	reqRegist.Password = self.password.text;
	Net.me:Send2(reqRegist);
end

File: trunk/Client/Assets/Script/Panels/PanelLogon.cs
using UnityEngine;
using System.Collections;
using Assets.Script.Frame;
using System.Reflection;
using System;
using UnityEngine.UI;

public class PanelLogon : PanelBase{

    InputField username;
    InputField password;
    public override void Init()
    {
        username = controls.GetInputField("InputFieldUsername");
        password = controls.GetInputField("InputFieldPassword");
        base.Init();
        
    }

    public void Awake()
    {
        Debug.Log("PanelLogon.Awake");
    }

    // Use this for initialization
    void Start () {
	
	}
	
	// Update is called once per frame
	void Update () {
	
	}

    public void OnButtonLogonClick()
    {
        Debug.Log("OnButtonLogonClick");
        //Hide();
        //PanelManager.me.Get((int)PanelId.PanelMain).Show();
        username = controls.GetInputField("InputFieldUsername");
        password = controls.GetInputField("InputFieldPassword");
        ReqLogon reqLogon = new ReqLogon();
        reqLogon.Username = username.text;
        reqLogon.Password = password.text;
        //Net.me.Send(reqLogon);
        Net.me.Send2(reqLogon);
    }

    public void OnButtonRegistClick()
    {
        Debug.Log("OnButtonRegistClick");
       // PanelManager.me.Get((int)PanelId.PanelGetVIT).Show();
       //// PanelManager.me.Get((int)PanelId.PanelTeamInfo).Show();
       //// PanelManager.me.Get((int)PanelId.PanelLogon).Hide();
       // return;

        InputField username = controls.GetInputField("InputFieldUsername");
        InputField password = controls.GetInputField("InputFieldPassword");
        
        ReqRegist reqRegist = new ReqRegist();
        reqRegist.Username = username.text;
        reqRegist.Password = password.text;
        Net.me.Send(reqRegist);
    }
}
